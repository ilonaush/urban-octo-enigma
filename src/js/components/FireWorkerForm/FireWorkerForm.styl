.fireWorkerForm
  display flex
  flex-direction column
