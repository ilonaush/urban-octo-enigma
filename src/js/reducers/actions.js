export default {
    employWorker: function (worker) {
        return {
            type: 'EMPLOY_WORKER',
            worker: worker
        }
    },
    editWorkTime: function (worker) {
        return {
            type: 'EDIT_WORK_TIME',
            worker: worker,
        }
    },
    fireWorker: function (workerId) {
        return {
            type: 'FIRE_WORKER',
            id: workerId
        }
    }
};
