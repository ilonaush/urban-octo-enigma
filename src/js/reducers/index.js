import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer   from './reducers';
import  { saveStore, loadStore } from '../services/LocalStorageService';

const predefinedStore = loadStore() || {};

const store = createStore(
    reducer,
    predefinedStore

);

store.subscribe(() => {
    saveStore(store)
});

export {
    store, predefinedStore
};
