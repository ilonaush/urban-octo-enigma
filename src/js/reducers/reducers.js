import {ACTIONS_TYPES} from "../constants";

const initialState = {
  loading: false,
};

let reducer = (state = initialState, action) => {
  switch (action.type) {
    case ACTIONS_TYPES.GET_CATS:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.ADD_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.FEED_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.HUG_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.WASH_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.ISSUE_CAT:
      return {
        ...state,
        cats: action.payload.cats,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      };
    case ACTIONS_TYPES.SET_LOADING: {
      return {
        ...state,
        loading: action.payload.loading,
        loadingType: action.payload.loadingType
      }
    }
    default:
      return state;
  }
};

export default reducer;
