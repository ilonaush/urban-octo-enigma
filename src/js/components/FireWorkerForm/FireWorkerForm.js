import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import actions from '../../reducers/actions';
import { bindActionCreators } from "redux";
import {Query} from "../../services/RequestService";
import "../AddWorkerForm/AddWorkerForm.styl";
import {withRouter} from "react-router-dom";

export class FireWorkerForm extends Component {

    constructor(props) {
        super(props);
        this.state = {
           worker: ''
        };

        // this.handleChange = this.handleChange.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
        this.handleSelect = this.handleSelect.bind(this);
        this.handleLoad = this.handleLoad.bind(this);

    }

    onSubmit(e) {
        e.preventDefault();
        debugger;

        this.props.actions.fireWorker(this.state.worker);
        const response = this.handleLoad();
    }

    handleSelect({target = null}) {
        debugger;
        console.log(target.value);
        this.setState({
            worker: target.value
        })
    }

    async handleLoad() {
        this.props.handleLoading(true);
        const response = await Query();
        if (response) {
            this.props.handleLoading(false);
            this.props.history.push('/');
        }
    }

    render() {
        return (
            <div>
                <form className='addWorkerForm'>
                    <select onChange={this.handleSelect}>
                        <option value=" ">Оберіть</option>
                        {this.props.workers.map((worker) =>
                            <option value={worker.id} >{worker.fullname}</option>
                        )}
                    </select>
                    <textarea placeholder='Describe reason please'/>
                    <button type='submit' onClick={this.onSubmit}>Fire a worker</button>
                </form>
            </div>
        );
    }
}

FireWorkerForm.propTypes = {};


function mapStateToProps(state) {
    return { workers: state.workers  }
}
function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators(actions, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(FireWorkerForm);
