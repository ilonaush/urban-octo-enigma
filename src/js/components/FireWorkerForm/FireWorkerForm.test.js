import React from 'react';
import FireWorkerForm from './FireWorkerForm';
import {mount, shallow} from "enzyme/build";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {store} from "../../reducers";

describe('FireWorkerForm', () => {
    it('renders correctly', () => {
        const FireWorkerFormComponent = renderer.create(<FireWorkerForm/>);
        expect((FireWorkerFormComponent).toJSON()).toMatchSnapshot();
    });

    it('changes its state on select', () => {
        const FireWorkerFormComponent = mount(shallow(
            <Provider store={store}>
                <FireWorkerForm/>
            </Provider>
        ).get(0));
        console.log(FireWorkerFormComponent.state());
        expect(FireWorkerFormComponent.state().workerID).toEqual('');
        FireWorkerFormComponent.find('select').simulate('change', {target: { value : '123'}});
        expect(FireWorkerFormComponent.state().workerID).toEqual('123');
    });
});
