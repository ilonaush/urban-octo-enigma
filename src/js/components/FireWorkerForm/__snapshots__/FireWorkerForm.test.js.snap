// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FireWorkerForm renders correctly 1`] = `
<form
  className="fireWorkerForm"
  onSubmit={[Function]}
>
  <select
    onChange={[Function]}
  >
    <option
      value=" "
    >
      Choose the meowy
    </option>
  </select>
  <textarea
    placeholder="Describe reason please"
  />
  <button
    id="submit-btn"
    type="submit"
  >
    Find home for the cat
  </button>
</form>
`;
